src/EnergyPlus/AirflowNetwork/src/Elements.cpp

```diff
@@ -3690,6 +3681,1015 @@ namespace AirflowNetwork {
+
+                for (j = 1; j <= (k - 1); ++j) {
+                    BetaStT(j) = (RhoStT(j + 1) - RhoStT(j)) / (HSt(j + 1) - HSt(j));
+                }
+            }
+
+            // CALCULATE STACK PRESSURE FOR THE PATH ITSELF for different flow directions
+            H = double(state.afn->AirflowNetworkLinkageData(i).NodeHeights[1]) - double(state.afn->AirflowNetworkLinkageData(i).NodeHeights[0]);
+            if (ll == 0 || ll == 3 || ll == 6) {
+                H -= state.afn->AirflowNetworkNodeData(From).NodeHeight;
+            }
+            if (ll < 3) {
+                H += state.afn->AirflowNetworkNodeData(To).NodeHeight;
+            }
+
+            // IF AIR FLOWS from "From" to "To"
+            Pref = Pbz + PzFrom + DpF(1);
+            DpP = psz(Pref, RhoLd(1), 0.0, 0.0, H, G);
+            DpL(i, 1) = (DpF(1) - DpT(1) + DpP);
+
+            // IF AIR FLOWS from "To" to "From"
+            Pref = Pbz + PzTo + DpT(1);
+            DpP = -psz(Pref, RhoLd(2), 0.0, 0.0, -H, G);
+            DpL(i, 2) = (DpF(1) - DpT(1) + DpP);
+
+            if (Ltyp == iComponentTypeNum::DOP) {
+                Pprof = OpenNum * (NrInt + 2);
+                presprofile(state, i, Pprof, G, DpF, DpT, BetaStF, BetaStT, RhoStF, RhoStT, From, To, ActLh, Hfl(i));
+                ++OpenNum;
+            }
+        }
+    }
+
+    Real64 DetailedOpeningSolver::psz(Real64 const Pz0,  // Pressure at altitude z0 [Pa]
+                                      Real64 const Rho0, // density at altitude z0 [kg/m3]
+                                      Real64 const beta, // density gradient [kg/m4]
+                                      Real64 const z0,   // reference altitude [m]
+                                      Real64 const z,    // altitude[m]
+                                      Real64 const g     // gravity field strength [N/kg]
+    )
+    {
+
+        // FUNCTION INFORMATION:
+        //       AUTHOR         Lixing Gu
+        //       DATE WRITTEN   Oct. 2005
+        //       MODIFIED       na
+        //       RE-ENGINEERED  This subroutine is revised based on psz function from COMIS
+
+        // PURPOSE OF THIS SUBROUTINE:
+        // This function determines the pressure due to buoyancy in a stratified density environment
+
+        // METHODOLOGY EMPLOYED:
+        // na
+
+        // REFERENCES:
+        // na
+
+        // USE STATEMENTS:
+        // na
+
+        // Return value
+        Real64 psz;
+
+        // Locals
+        // SUBROUTINE ARGUMENT DEFINITIONS:
+
+        // SUBROUTINE PARAMETER DEFINITIONS:
+        // na
+
+        // INTERFACE BLOCK SPECIFICATIONS
+        // na
+
+        // DERIVED TYPE DEFINITIONS
+        // na
+
+        // SUBROUTINE LOCAL VARIABLE DECLARATIONS:
+        Real64 dz;
+        Real64 rho;
+
+        dz = z - z0;
+        rho = (Rho0 + beta * dz / 2.0);
+        psz = -Pz0 * (1.0 - std::exp(-dz * rho * g / Pz0)); // Differential pressure from z to z0 [Pa]
```

Comment:
Can just return this expression without the temp variable.

Reply:
I believe this was a F77 function based on the naming.